src/convertToObject.js

```diff
@@ -6,7 +6,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesObject = {};
+  const declarations = sourceString.split(';');
+
+  declarations.forEach((declaration) => {
+    const [property, ...valueParts] = declaration.split(':');
+
+    if (!property || valueParts.length === 0) {
+      return;
+    }
+
+    const value = valueParts.join(':');
+
+    const trimmedProperty = property.trim();
+    const trimmedValue = value.trim();
+
+    if (trimmedProperty === '' || trimmedValue === '') {
+      return;
+    }
```

Comment:
[CODE PRACTICE] - Instead of using a forEach loop with a conditional return to filter out invalid declarations, consider using array methods like 'filter' to create a new array with valid declarations only. Then, use 'forEach' to iterate over the filtered array and construct the styles object.